tools/autograd/gen_python_functions.py

```diff
@@ -663,6 +663,7 @@ def parse_binding_arg(name):
             'value': argname,
             'formal': get_cpp_formal(tensor_options_arg),
         }
+        set_requires_grad = '.set_requires_grad(options.requires_grad())'
```

Comment:
This function has been rewritten - please do a rebase.
I think you can simply remove the `if` part here: https://github.com/pytorch/pytorch/blob/master/tools/autograd/gen_python_functions.py#L925-L926

Reply:
I mean removing `and not has_tensor_options(f)` from the `if` - you still want to gate it with `ps.tensor_options_args`.

And I'm not entirely sure whether `requires_grad` always exists when the python schema has any tensor options fields - the generation of the `requires_grad` is gated by different logic compared to other tensor options field: https://github.com/pytorch/pytorch/blob/master/tools/codegen/api/python.py#L563

So you probably need check whether `requires_grad` exists at all (and always apply it if it exists).
```
        set_requires_grad = f'.set_requires_grad({parser_outputs["requires_grad"].expr})' \
            if 'requires_grad' in parser_outputs else ''
```

Reply:
Thanks for the help @ljk53 !